android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -392,21 +391,41 @@ constructor(
     defaultRepository.addOrUpdate(resource = questionnaireResponse)
   }
 
-  suspend fun performExtraction(
+  fun performExtraction(
     context: Context,
     questionnaire: Questionnaire,
     questionnaireResponse: QuestionnaireResponse
   ): Bundle {
-
-    return ResourceMapper.extract(
-      questionnaire = questionnaire,
-      questionnaireResponse = questionnaireResponse,
-      StructureMapExtractionContext(
-        context = context,
-        transformSupportServices = transformSupportServices,
-        structureMapProvider = retrieveStructureMapProvider()
-      )
-    )
+    var bundle = Bundle()
+    viewModelScope.launch(dispatcherProvider.main()) {
+      kotlin
+        .runCatching {
+          withContext(dispatcherProvider.io()) {
+            ResourceMapper.extract(
+              questionnaire = questionnaire,
+              questionnaireResponse = questionnaireResponse,
+              StructureMapExtractionContext(
+                context = context,
+                transformSupportServices = transformSupportServices,
+                structureMapProvider = retrieveStructureMapProvider()
+              )
+            )
+          }
+        }
+        .onSuccess {
+          bundle = it
+          context.showToast(context.getString(R.string.structure_success))
+        }
+        .onFailure { exception ->
+          Timber.e(exception)
+          if (exception.message!!.contains("StructureMap")) {
```

Comment:
what's the exception that gets thrown here? Is there a more precise way to capture this exception? Since it's only checking for a string, couldn't this catch inappropriate exceptions

Reply:
`java.lang.NullPointerException:Attempt to invoke StructureMap on null object reference.`

Reply:
Given the above, added a check to verify that exception is an instance of `NullPointerException`